### ISSUE TITLE
VS2022

### ISSUE THREAD

**Issue by Anton-Interon** [zpqrtbnk/Zbu.ModelsBuilder#258]
Good Day,

Do you perhaps have an idea when Models Builder Extention for VS2022 will be made available ?

**Comment by zpqrtbnk** [comment]
Hey,

We do not have a team of active maintainers for Models Builder at the moment and therefore work on porting the extension to VS2022 has not started. VS2022 requires a separate build for the extensions, because of some changes in extensions management. As such, we cannot tell if nor when the VS2022 extension will be available.

I have done the job for my own [SolutionBook](https://github.com/zpqrtbnk/solutionbook) personal extension a while ago and it may be used as inspiration. It is not horribly complex, just annoying. But, we need someone with a bit of time to volunteer.

Anyone?

**Comment by zpqrtbnk** [comment]
By popular demand ;)

The issue with Visual Studio 2022 is that the Extensions mechanisms have changed enough that a single VSIX cannot support both Visual Studio 2019 and Visual Studio 2022. Therefore, a different VSIX has to be produced for Visual Studio 2022. For instance, for my [SolutionBook](https://github.com/zpqrtbnk/solutionbook) extension, I produce `SolutionBook.2019.vsix` and `SolutionBook.2022.vsix`.

Microsoft provides a [comprehensive documentation](https://docs.microsoft.com/en-us/visualstudio/extensibility/migration/update-visual-studio-extension?view=vs-2022) about the upgrade process. I have done this for [SolutionBook](https://github.com/zpqrtbnk/solutionbook) in July 2021 and it spans over less than a dozen commits, and it should not be *too* hard to follow.

For anyone wanting to give it a try, here is how I would do it. I would add two projects to the solution: `Our.ModelsBuilder.Extension.2019` and `Our.ModelsBuilder.Extension.2022`. I would move everything VSIX-related from `Our.ModelsBuilder.Extension` into `Our.ModelsBuilder.Extension.2019`, only leaving in `Our.ModelsBuilder.Extension` the bits that are Visual Studio version-agnostic. Then verify that the solution can still generate the correct VSIX for Visual Studio 2019. Then, populate the `.2022` project so that it can generate the new VSIX for Visual Studio 2022.

The trick is in getting things right, putting the right dependencies in each project, etc. And this is where copying from [SolutionBook](https://github.com/zpqrtbnk/solutionbook) may be a quick way to do it. Hint: I did [SolutionBook](https://github.com/zpqrtbnk/solutionbook) by copying from others.

I unfortunately don't have the bandwidth to deal with this now (or with the other issues that have accumulated...) but this is an administrative task that does not require any real nor deep knowledge of ModelsBuilder, C#, anything really. It's just moving files around. Don't be scared.

Anyone?

**Comment by jamiehowarth0** [comment]
@zpqrtbnk I'm having a crack at this now. Hoping it's not too difficult.

**Comment by benehudson** [comment]
@jamiehowarth0 @zpqrtbnk  following. I badly need a vsix for 2022. I also see modelsbuilder isn't ready for Umbraco V9.  Is there anyone on Umbraco staff that could pickup Stephan's work?  Modelsbuilder is crucial!!!  I was ready to start a new V9 project in 2 weeks, but without modelsbuilder I'm going to pump the brakes on that.

**Comment by motohead400** [comment]
Any chance anyone has taken this on? @jamiehowarth0 were you able to get this to work?

**Comment by nikro412956** [comment]
@jamiehowarth0 @zpqrtbnk @motohead400 just a ping to see if anything has changed.
